### ISSUE TITLE
[mruby-2.0] Bug with the Safe Navigation Operator (&.)

### ISSUE THREAD

**Issue by katzer** [mruby/mruby#4091]
@matz Since you have merged the mruby2-draft branch into master, there's an issue with the (&.) operator.

With mruby-1.4.1 you will get `nil` obviously:

```ruby
nil&.include? 123
# => nil
```

With mruby-head you will get an empty array:

```ruby
nil&.include? 123
# => []
```

You will get an empty array whatever you will do:

```ruby
nil&.first
# => []
```

Especially if you use the operator 2 times in a method, it returns an array of array and so one:

```ruby
nil&.include? 123
# => []
nil&.first
# => [[]]
```

**Comment by matz** [comment]
Hmm, I cannot reproduce the problem. Need more info.

**Comment by katzer** [comment]
The effected gem is [mruby-shelf-deflater](https://travis-ci.org/katzer/mruby-shelf-deflater). No other gem of mine has an issue with mruby-head.

I am testing on MacOS 10.13.6, however travis shoes the same issues on Ubuntu.

I noticed that by using the `&.` operator it always returns the result of the last method invocation (within an array).

Steps to reproduce:

1. `git clone https://github.com/katzer/mruby-shelf-deflater.git && cd mruby-shelf-deflater`
2. `MRUBY_VERSION=head rake compile`
3. Now you can add some code to `mrblib/deflater.rb`. You can use any method, but lets use `initialize` here.

   ```ruby
   def initialize(app, options = {})
     # [...]
     @compressible_types = options[:include]
     puts nil&.first.inspect
     puts nil&.first.inspect
     puts '----'
     puts nil&.first.inspect
     puts nil&.first.inspect
   end
   ```
4. `rake test`

   ```
   include
   [:include]
   ----
   ["----"]
   [["----"]]
   ```

**Comment by matz** [comment]
Fixed in 20c0c63. Thank you for the report.

**Comment by katzer** [comment]
LGTM
